### PR TITLE
fixed editor memory leak exception and 'must be invoked on debugger thread' exception

### PR DESCRIPTION
**Pull request by hza**
Meanwhile I am experimenting on syntax:

Screenshot 2023-07-23 at 20 49 58

but for java it will require import list from inner-tool and class+some method.

Screenshot 2023-07-23 at 20 50 52

### CODE REVIEW

File: src/main/java/util/debugger/NodeComponents.java [hunk 1]
```diff
@@ -97,45 +99,24 @@ public NodeComponents(XValueNodeImpl node) throws StackFrameThreadException, Eva
         this.debuggerContext = DebuggerManagerEx.getInstanceEx(this.project).getContext();
 
         this.frameProxy = this.evalContext.getFrameProxy();
-        this.thread = (ThreadReference) this.frameProxy
-                .threadProxy().getObjectReference();
+        this.thread = invokeOnDebuggerThread(() -> (ThreadReference) this.frameProxy
+                .threadProxy().getObjectReference(), this.evalContext);
```
Comment (hza) [inline src/main/java/util/debugger/NodeComponents.java:103]: invoking on debugger thread

File: src/main/java/ui/dialog/CopyValueDialog.java [hunk 2]
```diff
@@ -89,5 +89,11 @@ private void handleUpdate(Settings settings) {
         textArea.reload(settings);
         settingsChanged = true;
     }
+
+    @Override
+    protected void dispose() {
+        super.dispose();
+        textArea.dispose();
```
Comment (hza) [inline src/main/java/ui/dialog/CopyValueDialog.java:96]: memory leak without this

**Comment by chocovon** [comment]
Why would you need syntax when codes are all generated? For one thing, the generated code may naturally be wrong because of private fields or something else; for another, IMHO, the code display shouldn't be editable. If you need to use the code, edit it in the standard code editor could be more efficient.

**Comment by hza** [comment]
> Why would you need syntax when codes are all generated? For one thing, the generated code may naturally be wrong because of private fields or something else; for another, IMHO, the code display shouldn't be editable. If you need to use the code, edit it in the standard code editor could be more efficient.

I wanted code folding and have started on experimenting with JSON and it is requiring syntax checking. Now, for JSON it is done, including coding formatting using intellij formatter (in the next pull request):

Screenshot 2023-07-24 at 07 47 53

Syntax and folding for java are requiring import+class+method.

**Pull request merged**